### ISSUE TITLE
[auto/js] Concurrency issue with bulk config set

### ISSUE THREAD

**Issue by lukehoban** [pulumi/pulumi#6050]
The recent PR to parallelize `pulumi config set` operations unfortunately will lead to corrupting config (losing writes to the config file).  We'll need to revert that and likely invest in true support for bulk config set in the CLI.

Although we did fix the issue with concurrent writes to `~/.pulumi/credentials` zero-ing out the file, the `pulumi config set` still does write to disk in the `Pulumi.<stackname>.yaml` file, and this is not atomic - if two of these operations are run in parallel, the read of the file and write of the file by the two processes might be such that one of the writes is lost.  

This does point to a deeper concern, as users *could* hit these issues via their own scripting of the CLI.  But we need to make sure at least that Automation API doesn't directly trigger these problems.  

Two possible longer term solutions here:
1. Add a bulk config set operation in the CLI 
2. Ensure that `pulumi config set` is atomic (likely through a lock file and retries?)

**Comment by komalali** [comment]
I'd opt for the bulk config set via CLI - I've seen it come up a number of times from users as a nice-to-have.

**Comment by lukehoban** [comment]
The lack of this feature has somewhat significant performance impact.  For a program I have, 5 calls to `SetConfig` takes 10s wallclock time.

**Comment by komalali** [comment]
@lukehoban I agree. I think we're waiting on a design recommendation from you for a bulk config set and remove.

In addition, once we solve pulumi/pulumi#6062 we won't have the additional `select` call each time, which should help.
